config.go

```diff
@@ -228,73 +187,48 @@ func CreateDefaultConfigFile(configFilePath string) error {
 		return fmt.Errorf("failed to write headline to config file")
 	}
 
-	cfg := MinValuableConfig{
-		IOMode: "http",
-	}
-
-	cfg.ApplyEnv()
-
-	err = toml.NewEncoder(f).Encode(&cfg)
+	err = toml.NewEncoder(f).Encode(mvc)
 	if err != nil {
 		return fmt.Errorf("failed to encode config to file")
 	}
 
-	log.Infof("generated minimum valuable config to %s", configFilePath)
-
 	return nil
 }
 
-func (fm *Frontman) Initialize() error {
-	if fm.HubProxy != "" {
-		if !strings.HasPrefix(fm.HubProxy, "http") {
-			fm.HubProxy = "http://" + fm.HubProxy
-		}
-		_, err := url.Parse(fm.HubProxy)
-		if err != nil {
-			return fmt.Errorf("Failed to parse 'hub_proxy' URL")
+func (cfg *Config) validate() error {
+	if cfg.HubProxy != "" {
+		if !strings.HasPrefix(cfg.HubProxy, "http") {
+			cfg.HubProxy = "http://" + cfg.HubProxy
 		}
-	}
 
-	if fm.LogFile != "" {
-		err := addLogFileHook(fm.LogFile, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
-		if err != nil {
-			log.Error("Can't write logs to file: ", err.Error())
+		if _, err := url.Parse(cfg.HubProxy); err != nil {
+			return fmt.Errorf("failed to parse 'hub_proxy' URL")
 		}
 	}
 
-	if fm.LogSyslog != "" {
-		err := addSyslogHook(fm.LogSyslog)
-		if err != nil {
-			log.Error("Can't set up syslog: ", err.Error())
-		}
-	}
-
-	if fm.LogLevel != "" {
-		log.SetLevel(fm.LogLevel.LogrusLevel())
-	}
-
 	return nil
 }
 
-// HandleConfig configures Frontman with parameters specified in file
+// HandleAllConfigSetup prepares config for Frontman with parameters specified in file
 // if config file not exists default one created in form of MinValuableConfig
-// todo: this function must be removed once config is separated from Frontman
-func HandleConfig(fm *Frontman, configFilePath string) error {
-	err := fm.ReadConfigFromFile(configFilePath)
+func HandleAllConfigSetup(configFilePath string) (*Config, error) {
+	cfg := NewConfig()
+
+	err := TryUpdateConfigFromFile(cfg, configFilePath)
 	if os.IsNotExist(err) {
-		// this is ok
-		err = CreateDefaultConfigFile(configFilePath)
-		if err != nil {
-			log.Fatal(err)
+		mvc := NewMinimumConfig()
+		if err = GenerateDefaultConfigFile(mvc, configFilePath); err != nil {
+			return nil, err
 		}
+
+		cfg.MinValuableConfig = *mvc
 	} else if err != nil {
-		if strings.Contains(err.Error(), "cannot load TOML value of type int64 into a Go float") {
```

Comment:
This is was a very useful error description.

TOML lib we are using is very strict about float types and doesn't accept integer values and this is totally not clear!

E.g. if user set `sleep = 30` it will produce `cannot load TOML value of type int64 into a Go float`. I think this is not a kind of error that end user should face

Reply:
I see,
Does it make sense to return formatted error instead of doing log?

Reply:
Yes, please. 
And maybe you will also return the context to the error? Because `HandleAllConfigSetup` is doing multiple things it can be useful to describe which action produced the error, e.g. `fmt.Errorf("Config load error: %s", err.Error())`

Reply:
perfect, thanks